src/java.desktop/share/classes/javax/swing/plaf/synth/SynthGraphicsUtils.java

```diff
@@ -380,10 +391,18 @@ public void paintText(SynthContext ss, Graphics g, String text,
         FontMetrics fm = SwingUtilities2.getFontMetrics(c, g);
         Insets insets = SynthLookAndFeel.getPaintingInsets(ss, paintInsets);
 
-        paintViewR.x = insets.left;
-        paintViewR.y = insets.top;
-        paintViewR.width = c.getWidth() - (insets.left + insets.right);
-        paintViewR.height = c.getHeight() - (insets.top + insets.bottom);
+        final View v = (View)c.getClientProperty(BasicHTML.propertyKey);
```

Comment:
I guess this issue is only for JButton so is it needed to apply this for all components or we need to put a check for `c instance of JButton`

Reply:
Wasn't completely sure if it was necessary to add the check since it passed without, meaning I wouldn't need to import JButton into this class. However, it does make sense to check for JButtons, so I made the change. Tested the change and everything still passes.